packages/flutter/test/material/tab_assertions_test.dart

```diff
@@ -0,0 +1,14 @@
+import 'package:flutter/material.dart';
+import 'package:flutter_test/flutter_test.dart';
+
+void main() {
+  test('Tab throws clear error when both text and child are set', () {
+    // Wrap in a closure so the assertion is checked at runtime
+    expect(
+      () {
+        Tab(text: 'Hi', child: Text('World')); // no const
+      },
+      throwsA(isA<AssertionError>()),
```

Comment:
It would be better if we can also check the assertion error message. Something like:
```
  test('Tab throws clear error when both text and child are set', () {
    // Wrap in a closure so the assertion is checked at runtime
    expect(
      () {
        Tab(text: 'Hi', child: const Text('World')); // no const
      },
      throwsA(
        const TypeMatcher<AssertionError>().having(
          (AssertionError error) => error.message,
          'message',
          contains('Provide either text or child, not both, when creating a Tab.'),
        ),
      ),
    );
  });
```